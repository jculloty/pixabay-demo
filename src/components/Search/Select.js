import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';

import Label from './Label';

class Select extends PureComponent {
  static propTypes = {
    config: PropTypes.object.isRequired,
    value: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
  };

  handleChange = (event) => {
    const { name } = this.props.config;
    this.props.handleChange(name, event.target.value);
  }

  render() {
    const { options: list, default: defaultOption, name } = this.props.config;
    const options = list.sort().map((item) => <option value={item} key={item}>{item}</option>);

    if (defaultOption === undefined) {
      options.unshift(<option value="" key="_balnk_"></option>);
    }

    return (
      <div className="form-group">
        <Label name={name} />
        <select id={name} value={this.props.value} onChange={this.handleChange} className="custom-select">
          {options}
        </select>
      </div>
    );
  }
}

export default Select;
