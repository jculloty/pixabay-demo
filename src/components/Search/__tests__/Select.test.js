import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'test-utils/setup-enzyme';
import Select from '../Select';

const buildComponent = function(props = {}) {
  props = {
    ...{
      config: {
        name: 'testing',
        options: ['one', 'two', 'three'],
        default: 'one',
      },
      value: 'two',
      handleChange: () => {},
    },
    ...props,
  };

  return (<Select {...props} />);
};

describe('Search: Select', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders blank options when there is no default', () => {
    const props = {
      config: {
        name: 'testing',
        options: ['one', 'two', 'three'],
        default: undefined,
      },
    };
    const tree = renderer
      .create(buildComponent(props))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Typing triggeres the correct handler', () => {
    const mockChange = jest.fn();
    const wrapper = shallow(buildComponent({ handleChange: mockChange }));
    const expectedValue = 'three';

    wrapper.find('select').simulate('change', { target: { value: expectedValue } });

    expect(mockChange).toHaveBeenCalledTimes(1);
    expect(mockChange).toHaveBeenCalledWith('testing', expectedValue);
  });
});
