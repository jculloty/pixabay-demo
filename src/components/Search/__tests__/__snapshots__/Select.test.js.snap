// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search: Select renders blank options when there is no default 1`] = `
<div>
  <label
    htmlFor="testing"
  >
    Testing
  </label>
  <select
    id="testing"
    onChange={[Function]}
    value="two"
  >
    <option
      value=""
    />
    <option
      value="one"
    >
      one
    </option>
    <option
      value="three"
    >
      three
    </option>
    <option
      value="two"
    >
      two
    </option>
  </select>
</div>
`;

exports[`Search: Select renders correctly 1`] = `
<div>
  <label
    htmlFor="testing"
  >
    Testing
  </label>
  <select
    id="testing"
    onChange={[Function]}
    value="two"
  >
    <option
      value="one"
    >
      one
    </option>
    <option
      value="three"
    >
      three
    </option>
    <option
      value="two"
    >
      two
    </option>
  </select>
</div>
`;
